streams/src/test/java/org/apache/kafka/streams/state/internals/CachingWindowStoreTest.java

```diff
@@ -83,6 +96,104 @@ public void closeStore() {
         cachingStore.close();
     }
 
+    @Test
+    public void shouldTest() {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final StoreBuilder<WindowStore<String, String>> storeBuilder = Stores.windowStoreBuilder(
+                Stores.persistentWindowStore("store-name", 3600000L, 60000L, false),
+                Serdes.String(),
+                Serdes.String())
+                .withCachingEnabled();
+
+        builder.addStateStore(storeBuilder);
+
+        builder.stream(topic, Consumed.with(Serdes.String(), Serdes.String()))
+                .transform(() -> new Transformer<String, String, KeyValue<String, String>>() {
+
+                    private WindowStore<String, String> store;
+                    private int numRecordsProcessed;
+                    @Override
+                    public void init(final ProcessorContext processorContext) {
+                        this.store = (WindowStore<String, String>) processorContext.getStateStore("store-name");
+                        int count = 0;
+
+                        final KeyValueIterator<Windowed<String>, String> all = store.all();
+                        while (all.hasNext()) {
+                            count++;
+                            all.next();
+                        }
+
+                        assertThat(count, equalTo(0));
+                    }
+
+                    @Override
+                    public KeyValue<String, String> transform(final String key, final String value) {
+                        int count = 0;
+
+                        final KeyValueIterator<Windowed<String>, String> all = store.all();
+                        while (all.hasNext()) {
+                            count++;
+                            all.next();
+                        }
+                        assertThat(count, equalTo(numRecordsProcessed));
+
+                        store.put(value, value);
+
+                        numRecordsProcessed++;
+
+                        return new KeyValue<>(key, value);
+                    }
+
+                    @Override
+                    public void close() {
+
+                    }
+                }, "store-name");
+
+        final String bootstrapServers = "localhost:9092";
+        final Properties streamsConfiguration = new Properties();
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "test-app");
+        // Where to find Kafka broker(s).
```

Comment:
nit: do we need this comments? (similar below)

Reply:
ack